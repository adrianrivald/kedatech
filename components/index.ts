export * from "./Layout"
export * from "./Hero/Hero"