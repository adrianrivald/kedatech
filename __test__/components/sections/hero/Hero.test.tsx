import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import renderer from "react-test-renderer";
import { Hero } from "../../../../components";
import { useRouter } from "next/router";

it("Hero snapshot test", () => {
  const tree = renderer.create(<Hero />).toJSON();
  expect(tree).toMatchSnapshot();
});

beforeAll(() => {
  global.IntersectionObserver = class {
    constructor() {}
    observe() {}
    unobserve() {}
    disconnect() {}
  };
});

// Mock the next/router
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

(useRouter as jest.Mock).mockReturnValue({
  pathname: "/",
  query: {},
  push: jest.fn(),
  replace: jest.fn(),
  prefetch: jest.fn().mockResolvedValue(undefined),
  asPath: "/",
});

test("Hero test", () => {
  render(<Hero />);

  const heroHeadingEl = screen.getByRole("heroHeading");
  expect(heroHeadingEl).toHaveTextContent(
    "Biar Sistem yang Catat, Anda Fokus Berkembang."
  );

  const heroDescEl = screen.getByRole("heroDesc");
  expect(heroDescEl).toHaveTextContent(
    "Kelola stok, pantau arus barang, dan catat keuntungan harian secara otomatis dalam satu platform. Sistem ERP modern kami dirancang khusus untuk membantu pengusaha mengelola operasional bisnis dengan lebih efisien, tanpa perlu repot mencatat manual."
  );

  const heroButtonEl = screen.getByRole("heroButton");
  expect(heroButtonEl).toHaveTextContent("Konsultasikan");
});
