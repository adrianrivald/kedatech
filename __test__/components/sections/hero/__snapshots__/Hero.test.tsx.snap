// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero snapshot test 1`] = `
<div
  className="flex lg:flex-row flex-col-reverse justify-between items-center min-h-[400px] gap-4 lg:gap-16 p-4 lg:p-0 max-w-[100rem] mx-auto"
  id="about"
>
  <div
    className="css-0"
    style={
      {
        "animationDelay": "0ms",
        "opacity": 0,
      }
    }
  >
    <div
      className="lg:px-16 lg:max-w-[40rem]"
      id="our-words"
      style={{}}
    >
      <h1
        className="text-[32px] font-bold text-center lg:text-left"
        role="heroHeading"
      >
        Biar Sistem yang Catat, Anda Fokus Berkembang.
      </h1>
      <p
        className="mt-4 text-base text-center lg:text-left"
        role="heroDesc"
      >
        Kelola stok, pantau arus barang, dan catat keuntungan harian secara otomatis dalam satu platform. Sistem ERP modern kami dirancang khusus untuk membantu pengusaha mengelola operasional bisnis dengan lebih efisien, tanpa perlu repot mencatat manual.
      </p>
      <div
        className="flex flex-col lg:flex-row items-center gap-4 mt-8"
      >
        <button
          className="rounded-lg px-4 py-2 border-2 min-h-12 bg-primary-500 border-primary-500 text-white w-full lg:w-auto"
          onClick={[Function]}
          role="heroButton"
        >
          Konsultasikan
        </button>
      </div>
    </div>
  </div>
  <div
    className="w-full lg:w-1/2 css-0"
    style={
      {
        "animationDelay": "0ms",
        "opacity": 0,
      }
    }
  >
    <img
      alt="doctor"
      className="w-full"
      data-nimg="1"
      decoding="async"
      height={620}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fassets%2Fimages%2Fbg-hero.png&w=1200&q=75"
      srcSet="/_next/image?url=%2Fassets%2Fimages%2Fbg-hero.png&w=640&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fbg-hero.png&w=1200&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={591}
    />
  </div>
</div>
`;
